pkg/virt-controller/watch/migration.go

```diff
@@ -1107,6 +1150,13 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
 		return fmt.Errorf("vmi is inelgible for migration because another migration job is running")
 	}
 
+	if err = c.handleMigrationBackoff(key, vmi, migration, pod); err != nil {
```

Comment:
Wouldn't it be sufficient to list all matching migrations here and decide if we need to back off? (Using cluster view rather than internal state) We could track the back-off on the migration object. I think I am not really sure if it is valuable to emit an event/warning in this case rather than provide information that can be lookup on why the migration is taking so long... On first look, I think it might be even simpler. Did you consider this approach?

Reply:
Do you mean matching failed migrations for the VMI? I'd rather not, they get garbage collected so that's error-prone, having an internal count in memory it's more reliable I think, also I don't have to introduce any new API (new fields in the object).

About the event, well I don't know I actually think it's pretty helpful to get notified about these failures, on the same page as what kubernetes does when for example there's a loop crash backoff, Do you have a different way to communicate this failure in mind?

Reply:
I believe we garbage collect only  5+ oldest migrations. The benefit of having the field and not using the internal representation is that it is less likely to miss a case when you should remove the internal entries(e.g success)

Ok, on second thought with your example I like what we do here. (I mean the event)